libs/ledger/include/ledger/protocols/main_chain_rpc_service.hpp

```diff
@@ -169,11 +171,12 @@ class MainChainRpcService : public muddle::rpc::Server,
 
   /// @name State Machine Data
   /// @{
-  RpcClient       rpc_client_;
-  StateMachinePtr state_machine_;
-  Address         current_peer_address_;
-  BlockHash       current_missing_block_;
-  Promise         current_request_;
+  RpcClient             rpc_client_;
+  StateMachinePtr       state_machine_;
+  Address               current_peer_address_;
+  BlockHash             current_missing_block_;
+  Promise               current_request_;
+  std::atomic<uint16_t> loose_blocks_seen_{0};
```

Comment:
Count loose blocks seen and use this to guess that we are out of sync. There was a condition where the forward sync would request from a node, get Genesis or genesis + 1 block, and then stay in the syncronised state forever, dumping all loose blocks